fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/data/CollectionData.java

```diff
@@ -20,71 +20,27 @@
 
 import java.math.BigDecimal;
 import java.time.LocalDate;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
 
+@AllArgsConstructor
+@Getter
+@Setter
 public final class CollectionData {
 
-    private final BigDecimal availableDisbursementAmount;
-    private final int pastDueDays;
-    private final LocalDate nextPaymentDueDate;
-    private final int delinquentDays;
-    private final LocalDate delinquentDate;
-    private final BigDecimal delinquentAmount;
-    private final LocalDate lastPaymentDate;
-    private final BigDecimal lastPaymentAmount;
-
-    private CollectionData(BigDecimal availableDisbursementAmount, int pastDueDays, LocalDate nextPaymentDueDate, int delinquentDays,
-            LocalDate delinquentDate, BigDecimal delinquentAmount, LocalDate lastPaymentDate, BigDecimal lastPaymentAmount) {
-        this.availableDisbursementAmount = availableDisbursementAmount;
-        this.pastDueDays = pastDueDays;
-        this.nextPaymentDueDate = nextPaymentDueDate;
-        this.delinquentDays = delinquentDays;
-        this.delinquentDate = delinquentDate;
-        this.delinquentAmount = delinquentAmount;
-        this.lastPaymentDate = lastPaymentDate;
-        this.lastPaymentAmount = lastPaymentAmount;
-    }
-
-    public static CollectionData instance(BigDecimal availableDisbursementAmount, int pastDueDays, LocalDate nextPaymentDueDate,
-            int delinquentDays, LocalDate delinquentDate, BigDecimal delinquentAmount, LocalDate lastPaymentDate,
-            BigDecimal lastPaymentAmount) {
-        return new CollectionData(availableDisbursementAmount, pastDueDays, nextPaymentDueDate, delinquentDays, delinquentDate,
-                delinquentAmount, lastPaymentDate, lastPaymentAmount);
-    }
+    private BigDecimal availableDisbursementAmount;
+    private int pastDueDays;
```

Comment:
Change to long and we can avoid back and forth conversation between Integer and Long (int and long)

Reply:
Same as above, issue with the original JPQL select

Reply:
Using Long type